bindings/rust/src/lib.rs

```diff
@@ -34,4 +251,124 @@ mod tests {
+        assert!(instance.find_exported_function(&"tab").is_none());
+        assert!(instance.find_exported_function(&"mem").is_none());
+    }
+
+    #[test]
+    fn unsafe_execute_wasm() {
+        /* wat2wasm
+          (func)
+          (func (result i32) i32.const 42)
+          (func (param i32 i32) (result i32)
+            (i32.div_u (local.get 0) (local.get 1))
+          )
+          (func unreachable)
+        */
+        let input = hex::decode("0061736d01000000010e036000006000017f60027f7f017f030504000102000a150402000b0400412a0b0700200020016e0b0300000b").unwrap();
+        let module = parse(&input);
+        assert!(module.is_ok());
+        let instance = module.unwrap().instantiate();
+        assert!(instance.is_ok());
+        let mut instance = instance.unwrap();
+
+        let result = unsafe { instance.unsafe_execute(0, &[], 0) };
+        assert!(!result.trapped());
+        assert!(!result.value().is_some());
+
+        let result = unsafe { instance.unsafe_execute(1, &[], 0) };
+        assert!(!result.trapped());
+        assert!(result.value().is_some());
+        assert_eq!(result.value().unwrap().as_i32(), 42);
+
+        // Explicit type specification
+        let result =
+            unsafe { instance.unsafe_execute(2, &[(42 as i32).into(), (2 as i32).into()], 0) };
+        assert!(!result.trapped());
+        assert!(result.value().is_some());
+        assert_eq!(result.value().unwrap().as_i32(), 21);
+
+        // Implicit i64 types (even though the code expects i32)
+        let result = unsafe { instance.unsafe_execute(2, &[42.into(), 2.into()], 0) };
+        assert!(!result.trapped());
+        assert!(result.value().is_some());
+        assert_eq!(result.value().unwrap().as_i32(), 21);
+
+        let result = unsafe { instance.unsafe_execute(3, &[], 0) };
+        assert!(result.trapped());
+        assert!(!result.value().is_some());
+    }
+
+    #[test]
+    fn execute_wasm() {
+        /* wat2wasm
+        (module
+          (func $f (export "foo") (result i32) (i32.const 42))
+          (global (export "g1") i32 (i32.const 0))
+          (table (export "tab") 0 anyfunc)
+          (memory (export "mem") 1 2)
+        )
+        */
+        let input = hex::decode(
+        "0061736d010000000105016000017f030201000404017000000504010101020606017f0041000b07180403666f6f00000267310300037461620100036d656d02000a06010400412a0b").unwrap();
+
+        let module = parse(&input);
+        assert!(module.is_ok());
+        let instance = module.unwrap().instantiate();
+        assert!(instance.is_ok());
+        let mut instance = instance.unwrap();
+
+        let result = instance.execute("foo", &[], 0);
+        assert!(result.is_ok());
+        let result = result.unwrap();
+        assert!(!result.trapped());
+        assert!(result.value().is_some());
+        assert_eq!(result.value().unwrap().as_i32(), 42);
+
+        let result = instance.execute("bar", &[], 0);
+        assert!(result.is_err());
+
```

Comment:
You could add also an attempt to execute non-function export, like `g1`

Reply:
This functionality is moved to the next PR.